capa/rules/__init__.py

```diff
@@ -89,6 +90,46 @@ class Scope(str, Enum):
 # used only to specify supported features per scope.
 # not used to validate rules.
 GLOBAL_SCOPE = "global"
+DEV_SCOPE = "dev"
+
+
+# these literals are used to check if the flavor
+# of a rule is correct.
+STATIC_SCOPES = (
+    FILE_SCOPE,
+    GLOBAL_SCOPE,
+    FUNCTION_SCOPE,
+    BASIC_BLOCK_SCOPE,
+    INSTRUCTION_SCOPE,
+)
+DYNAMIC_SCOPES = (
+    FILE_SCOPE,
+    GLOBAL_SCOPE,
+    PROCESS_SCOPE,
+    THREAD_SCOPE,
+    DEV_SCOPE,
+)
+
+
+@dataclass
+class Scopes:
+    static: str
+    dynamic: str
+
+    def __eq__(self, scope) -> bool:
+        assert isinstance(scope, str) or isinstance(scope, Scope)
+        return (scope == self.static) or (scope == self.dynamic)
```

Comment:
im not comfortable overriding `__eq__` like this, because it allows very different types to appear to be the same, like `(Scopes("function", "thread") == "function") is True` which is pretty surprising.

perhaps you can introduce an accessor, like `has_scope(str) -> bool` to express this query?

Reply:
I see, that's understandable.

However, before moving forward with the `has_scope()` method, I'd like to propose overriding the _in_ operator, which would result in  something like this: `if "function" in rules.scopes:`, etc. But if you prefer `has_scopes()` method to this then let me know, since both are fine by me.

Reply:
`in` is a great idea!